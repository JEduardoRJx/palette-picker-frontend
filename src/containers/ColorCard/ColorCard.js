import React from 'react';
import './ColorCard.scss';
import locked from '../../images/locked.svg';
import unlocked from '../../images/unlocked.svg';

export const ColorCard = ({ colorInfo, toggleLock }) => {
  const { color, isLocked} = colorInfo;
  console.log("Color", color)
  const icon = isLocked ? locked : unlocked;
  const description = isLocked ? 'locked icon' : 'unlocked icon'
  
  return (
    <article className="color-card"
      style={{backgroundColor: color}}>
      <button className='icon-btn' 
        type='button'
        onClick={e => toggleLock(e)}>
        <img className='lock-icon' src={icon} alt={description} />
      <h3 className='color-text'>{color.toUpperCase()}</h3>
      </button>
    </article>
  )
}

export default ColorCard;