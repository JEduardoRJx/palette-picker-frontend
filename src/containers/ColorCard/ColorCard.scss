.color-card {
  background-color: blueviolet;
  border: 5px solid black;
  height: 100vh;
  width: 100vw;
  margin-top: 30px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.icon-btn {
  width: 100%;
  height: 15%;
  margin-top: 10%;
  background-color: transparent;
  border: none;
}

.lock-icon {
  width: 100%;
  height: 100%;
}

.color-text {
  font-size: 4em;
  color: white;
}