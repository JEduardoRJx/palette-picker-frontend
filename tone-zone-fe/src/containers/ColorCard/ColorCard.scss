.color-card {
  background-color: blueviolet;
  border: 5px solid black;
  height: 500px;
  width: 100vw;
}