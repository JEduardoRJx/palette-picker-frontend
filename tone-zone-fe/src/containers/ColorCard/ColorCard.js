import React from 'react';
import './ColorCard.scss';

export const ColorCard = () => {
  return (
    <div className="color-card">
      
    </div>
  )
}

export default ColorCard;