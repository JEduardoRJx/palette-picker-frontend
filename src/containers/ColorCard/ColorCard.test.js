import React from 'react';
import { shallow } from 'enzyme';
import ColorCard from './ColorCard'

describe('ColorCard', () => {
  const mockColorInfo = {
    color: "#cecccc",
    isLocked: false
  }

  it('should match the snapshot ', () => {
    const wrapper = shallow(<ColorCard
      colorInfo={mockColorInfo} 
      toggleLock={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});