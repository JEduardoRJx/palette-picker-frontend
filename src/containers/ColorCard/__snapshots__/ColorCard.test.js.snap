// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard should match the snapshot  1`] = `
<article
  className="color-card"
  style={
    Object {
      "backgroundColor": "#cecccc",
    }
  }
>
  <button
    className="icon-btn"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="unlocked icon"
      className="lock-icon"
      src="unlocked.svg"
    />
    <h3
      className="color-text"
    >
      #CECCCC
    </h3>
  </button>
</article>
`;
